### ISSUE TITLE
AMG filling method fails to converge for velocity fields

### ISSUE THREAD

**Issue by NoraLoose** [CWorthy-ocean/roms-tools#179]
The generation of the boundary forcing is really slow. In #157 I hypothesized that the underlying reason is that we do the regridding inefficiently. However, the real reason is *once again* (😄) the infilling method. While we achieved a significant performance boost by switching to the AMG filling method, we never tested it for **velocity fields** (only for wind fields, and 3D temperature). 

Here is an example for the AMG fill of temperature. The figure shows the original surface temperature on the left, and the filled surface temperature on the right. The timing (3s) is for the full 3D field (50 depth levels).
<img width="980" alt="Screenshot 2024-10-24 at 1 11 24 PM" src="https://github.com/user-attachments/assets/59c324ea-fcfe-4fb8-ac00-3f6b6e656dea">

In contrast, the AMG filling method has trouble converging for velocity fields. The method spends 80s filling in the 3D field, but the result (shown on the right) still shows NaNs over land:
<img width="972" alt="Screenshot 2024-10-24 at 1 17 37 PM" src="https://github.com/user-attachments/assets/6d9d7606-0f09-4135-a202-22faa91bf910">

TLDR, the structure of the data can make a huge difference for convergence. In hindsight, it makes sense that velocity is a difficult field to fill in, with its small-scale structure and many sign changes.

@sdbachman my suggestion is to simply fill velocity fields with zeros over land. Or can you think of a better solution?

**Comment by NoraLoose** [comment]
@sdbachman found the real underlying issue: it's not the structure of the velocity data, but the fact that GLORYS uses different masks for tracer vs. velocity fields. As a result, we need to set up two different solvers, one with the tracer mask, and one with the velocity mask. After implementing this, the issue is resolved: convergence is fast (~3s for 50 depth levels, same as for the tracer fields) and the result does not contain any NaNs.

<img width="972" alt="Screenshot 2024-10-24 at 2 55 16 PM" src="https://github.com/user-attachments/assets/61961648-9421-44f2-9c54-23c9bb43141c">
